### PR TITLE
Bump Doorkeeper Gem

### PR DESCRIPTION
**Pull request by zendesk-mattlefevre**
**Note**: Samson is a public repo, do not include Zendesk-internal information, urls, etc.

## Description
To keep up on known vulnerabilities, we are updating to the latest version of the Doorkeeper gem. 

### Risks
- Med: While only a minor version bump, the change does seem to change some logic we touch. We'll want to test this thoroughly.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ group :preload do
   gem 'pagy'
   gem 'audited'
   gem 'soft_deletion'
-  gem 'doorkeeper', '~> 5.4.0' # TODO: upgrade breaks `rails c`
+  gem 'doorkeeper', '~> 5.6.0' # TODO: upgrade breaks `rails c`
```
Comment (grosser) [inline Gemfile:18]:
```suggestion
  gem 'doorkeeper', '~> 5.6.0'
```

File: config/initializers/doorkeeper.rb [hunk 2]
```diff
@@ -11,15 +11,15 @@ module Doorkeeper
...
-  module Orm
-    module ActiveRecord
-      class << self
-        undef initialize_models!
-        def initialize_models!
-        end
-      end
-    end
-  end
+  # module Orm
```
Comment (grosser) [inline config/initializers/doorkeeper.rb:14]: should delete or explain why it's commented out

File: config/initializers/doorkeeper.rb [hunk 3]
```diff
@@ -11,15 +11,15 @@ module Doorkeeper
...
+  #   module ActiveRecord
+  #     class << self
+  #       # undef initialize_models!
+  #       def initialize_models!
+  #       end
+  #     end
+  #   end
+  # end
```
Comment (grosser) [inline config/initializers/doorkeeper.rb:22]: redo test explained above to make sure this is safe

File: config/initializers/doorkeeper.rb [hunk 4]
```diff
@@ -11,15 +11,6 @@ module Doorkeeper
...
   autoload :RedirectUriValidator, 'doorkeeper/orm/active_record/redirect_uri_validator'
-  module Orm
-    module ActiveRecord
-      class << self
-        undef initialize_models!
-        def initialize_models!
-        end
-      end
-    end
-  end
```
Comment (sarahjmiller) [inline config/initializers/doorkeeper.rb:22]: Just curious why this needed to be deleted.

**Review by sarahjmiller — APPROVED** [review]
Looks good to me! Approving for Deploy Experience team, although we may also want to be sure to get sign off from @grosser.

File: config/initializers/doorkeeper.rb (hunk 4 above)
Comment (zendesk-mattlefevre) [reply config/initializers/doorkeeper.rb:22]: This function no longer exists in the Doorkeeper gem, so we don't need to replace it with an empty method.

**Review by grosser — APPROVED** [review]

**Pull request merged**